File: ws/cuichan/U2/WS24-UnitTest/calculator/src/main/java/ec/espe/edu/calculator/controller/OperationController.java
package ec.espe.edu.calculator.controller;

import ec.espe.edu.calculator.model.BasicOperation;

/**
 *
 * @author LABS-DCCO
 */
public class OperationController {
    private BasicOperation basicOperation;
    
    public float add(float addend1,float addend2){
        float sum;
        sum = addend1 + addend2;
        return sum;
    }
    
    public float addition(BasicOperation basicOperation){
        float sum;
        sum = basicOperation.getTerm1() + basicOperation.getTerm2();
        return sum;
    }
    
    public OperationController(BasicOperation basicOperation) {
        this.basicOperation = basicOperation;
    }

    /**
     * @return the basicOperation
     */
    public BasicOperation getBasicOperation() {
        return basicOperation;
    }

    /**
     * @param basicOperation the basicOperation to set
     */
    public void setBasicOperation(BasicOperation basicOperation) {
        this.basicOperation = basicOperation;
    }
    
}

File: hw/gongora/U2/HW28UnitTestJS/unitTestJavascript/ec/edu/espe/controller/OperationController.js

class OperationController{
    #basicOperation

    constructor(basicOperation){
        this.#basicOperation = basicOperation;
    }

    add(addend1, addend2) {
        let sum = addend1 + addend2;
        return sum;
    }

    add() {
        let sum = this.#basicOperation.getTerm1() + this.#basicOperation.getTerm2();
        return sum;
    }

    getBasicOperation(){
        return this.#basicOperation;
    }

    setBasicOperation(basicOperation){
        this.#basicOperation = basicOperation
    }
}

module.exports = OperationController;